### PR TITLE
fix(utilities): skip invalid style attributes

### LINKED ISSUE #256: Throws error when styles are corrupted
## Expected Behavior

- Parse a string with a corrupted style attribute, e.g. ‘&lt;p style="font - size: 1em"&gt;X&lt;/p&gt;’
- Return a version that skips the corrupted style ‘&lt;p&gt;X&lt;/p&gt;’? 

## Steps to Reproduce

Throws an exception: undefined:1:6: property missing ':'
## Reproducible Demo
https://jsfiddle.net/bjwmd5s6/

## Environment

- Version: 1.2.6
- Platform: Mac
- Browser: Tested on Firefox

### PR DESCRIPTION
**Pull request by dogriffiths**
Fixes #256

## Checklist:

- [x] Tests
- [ ] Documentation
- [ ] Types

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -153,3 +153,14 @@ describe('trim option', () => {
     );
   });
 });
+
+describe('corrupted styles', () => {
```
Comment (remarkablemark) [inline test/index.test.js:157]: Do you mind renaming `corrupted` to `invalid`?

File: test/index.test.js [hunk 2]
```diff
@@ -153,3 +153,14 @@ describe('trim option', () => {
...
+  it('copes with corrupted styles', () => {
```
Comment (remarkablemark) [inline test/index.test.js:158]: Can you rename this description to:

```js
it('skips invalid styles', () => {
```

File: test/utilities.test.js [hunk 3]
```diff
@@ -117,4 +117,11 @@ describe('setStyleProp', () => {
     expect(setStyleProp(style, props)).toBe(undefined);
     expect(props).toMatchSnapshot();
   });
+
+  it('does not set props.style when style attribute corrupt', () => {
```
Comment (remarkablemark) [inline test/utilities.test.js:121]: Do you mind updating description to:

```js
it('does not set props.style when style attribute is invalid', () => {
```

**Review by remarkablemark — CHANGES_REQUESTED** [review]
@dogriffiths Thanks for opening this PR!

It looks good overall. I left some comments and can you fix lint errors?

```sh
npm run lint:fix
```

**Comment by remarkablemark** [comment]
Also, do you think the [README](https://github.com/remarkablemark/html-react-parser/blob/v1.2.6/README.md) should be updated?

**Comment by remarkablemark** [comment]
@dogriffiths would you be able to help get the build green so I can move this PR forward?

File: test/index.test.js (hunk 1 above)
Comment (dogriffiths) [reply test/index.test.js:157]: Sorry. Just saw this comment. Will redo PR now.

Comment (remarkablemark) [reply test/index.test.js]: Thanks!

**Review by remarkablemark — APPROVED** [review]
LGTM, thanks for working on this @dogriffiths!

**Comment by remarkablemark** [comment]
Published [v1.2.8](https://github.com/remarkablemark/html-react-parser/releases/tag/v1.2.8):

[npm](https://www.npmjs.com/package/html-react-parser):

```sh
npm i html-react-parser@1.2.8
```

[Yarn](https://yarnpkg.com/package/html-react-parser):

```sh
yarn add html-react-parser@1.2.8
```

**Pull request merged**